packages/webview/src/component/terminal/json-colorizer.ts

```diff
@@ -0,0 +1,258 @@
+
+    // Check if we're in a number context
+    for (let k = index - 1; k >= 0; k--) {
+      const prevChar = jsonLine[k];
+      if (/\s/.test(prevChar)) continue;
+      return prevChar === ':' || prevChar === '[' || prevChar === ',';
+    }
+    return false;
+  }
+
+  private processNumber(jsonLine: string, startIndex: number): ProcessResult {
+    let i = startIndex;
+    if (jsonLine[i] === '-') i++;
+
+    while (i < jsonLine.length && /[\d.]/.test(jsonLine[i])) {
+      i++;
+    }
+
+    const number = jsonLine.substring(startIndex, i);
+    const text = this.colorScheme.numberColor ? this.colorScheme.numberColor + number + this.colorScheme.reset : number;
+    return {
+      text,
+      nextIndex: i,
+    };
+  }
+
+  private processKeyword(jsonLine: string, startIndex: number): ProcessResult | undefined {
+    // Handle boolean values
+    if (jsonLine.substring(startIndex, startIndex + 4) === 'true') {
+      const text = this.colorScheme.booleanColor
+        ? this.colorScheme.booleanColor + 'true' + this.colorScheme.reset
+        : 'true';
+      return {
+        text,
+        nextIndex: startIndex + 4,
+      };
+    }
+
+    if (jsonLine.substring(startIndex, startIndex + 5) === 'false') {
+      const text = this.colorScheme.booleanColor
+        ? this.colorScheme.booleanColor + 'false' + this.colorScheme.reset
+        : 'false';
+      return {
+        text,
+        nextIndex: startIndex + 5,
+      };
+    }
+
+    // Handle null
+    if (jsonLine.substring(startIndex, startIndex + 4) === 'null') {
+      const text = this.colorScheme.nullColor ? this.colorScheme.nullColor + 'null' + this.colorScheme.reset : 'null';
+      return {
+        text,
+        nextIndex: startIndex + 4,
+      };
+    }
+
+    return undefined;
+  }
+}
+
+/**
+ * Checks if a string appears to be valid JSON.
+ * A valid JSON line should have both { and } and contain at least one key-value pair pattern.
+ *
+ * @param line - The line to check
+ * @returns true if the line appears to be JSON
+ */
+export function isValidJSON(line: string): boolean {
+  const trimmed = line.trim();
+  if (!trimmed) return false;
+
+  // Find positions of braces
+  const openBrace = trimmed.indexOf('{');
+  const closeBrace = trimmed.lastIndexOf('}');
+
+  // Must have both braces and { must come before }
+  if (openBrace === -1 || closeBrace === -1 || openBrace >= closeBrace) return false;
+
+  // Extract content between braces and check for key-value pair pattern
+  const contentBetweenBraces = trimmed.substring(openBrace + 1, closeBrace);
+
+  // Look for key-value pair pattern: "key": value or "key":value
+  const kvpPattern = /"[^"]+"\s*:\s*[^,}]+/;
+  return kvpPattern.test(contentBetweenBraces);
+}
+
+/**
+ * Detects if log lines are predominantly JSON format.
+ *
+ * @param lines - Array of log lines to analyze
+ * @returns true if logs should be treated as JSON format
+ */
+export function detectJsonLogs(lines: string[]): boolean {
```

Comment:
I would prefer to not include the json detection for the moment, and only have the possibility for the user to select it in the toolbar. We can optimize later with detection